scripting/src/main/java/org/infinispan/scripting/impl/DataTypedCache.java

```diff
@@ -69,22 +77,27 @@ public V get(Object key) {
 
    @Override
    public CacheSet<K> keySet() {
-      CacheStream<K> stream = getDelegate().keySet().stream().map(this::toDataType);
-      return new SetAsCacheSet<>(stream.collect(Collectors.toSet()));
+      CacheStream<K> stream = getDelegate().keySet().stream().map(
+            new ValueToTypedValueFunction<>(dataTypedCacheManager.dataType, dataTypedCacheManager.marshaller));
+      return new SetAsCacheSet<>(stream.collect(
```

Comment:
All of the examples of where we do that currently exist in the interceptor stack which rely on the cache to do alot of the other heavy lifting.

Unfortunately this means you would have to do it manually by implementing all of the various methods to do the proper conversion.

I am guessing you would also want to prevent users from doing a write to this keySet, or is that possible?